pkg/cluster/logsaarchive.go

```diff
@@ -0,0 +1,77 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package cluster
+
+import (
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/siderolabs/talos/pkg/imager/filemap"
+)
+
+// SaveClusterLogsArchive saves all logs from the cluster state directory to a gzip archive.
+func SaveClusterLogsArchive(statePath, archivePath string) {
```

Comment:
I think it's better create a separate version of this function that returns an error, and move all print-on-error calls here, inside this variant.

You could also wrap all closers inside sync.OnceValue and call them at the final `return cmp.Or(closer1(), closer2(),...)`